recipes/emsdk_installer/all/conanfile.py

```diff
@@ -0,0 +1,121 @@
+from conans import ConanFile, tools
+import os
+
+
+class EmSDKInstallerConan(ConanFile):
+    name = "emsdk_installer"
+    description = "Emscripten is an Open Source LLVM to JavaScript compiler"
+    url = "https://github.com/bincrafters/conan-emsdk_installer"
+    homepage = "https://github.com/kripken/emscripten"
+    topics = ("conan", "emsdk", "emscripten", "installer", "sdk")
+    license = "MIT"
+
+    short_paths = True
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "emsdk-%s" % self.version
+        os.rename(extracted_folder, self._source_subfolder)
+
```

Comment:
A `validate` method to check whether `os` is valid would be useful, I think.
This recipe is supposed to be used in a cross build configuration. So it should also probably test whether the target os is actually `wasm`.
Something like:
```python
    def validate(self):
        if self.settings.os not in ("Linux", "Macos", "Windows"):
             raise ConanInvalidConfiguration("Invalid OS. Only Linux, Macos and Windows are supported.")
        if hasattr(self, "settings_target"):
            if self.settings_target.os != "wasm":
                 raise ConanInvalidConfiguration("Emscripten targets only wasm, not {}.".format(self.settings_target.os)
```
You will also need to add `from conans import ConanInvalidConfiguration` at the top of this file.

Reply:
When providing "--profile:host" self.settings.os contains the value of that profile. So when the consumer sets a profile as host which os is Emscripten this does not work.

I think  "wasm" is not an os.

To be honest I do not know what configurations are wrong.
I think it is a great idea to give the user feedback.
Maybe we could use "self.settings_build" instead of "self.settings.os" to check if the build os is linux mac or windows.